conans/client/conan_api.py

```diff
@@ -72,14 +72,43 @@ def wrapper(*args, **kwargs):
     return wrapper
 
 
-def prepare_cwd(cwd):
-    if cwd:
-        if os.path.isabs(cwd):
-            return cwd
+def _make_abs_path(path, cwd=None, default=None):
+    """convert 'path' to absolute if necessary (could be already absolute)
+    if not defined (empty, or None), will return 'default' one or 'cwd'
+    """
+    cwd = cwd or os.getcwd()
+    if not path:
+        return default or cwd
+    if os.path.isabs(path):
+        return path
+    return os.path.normpath(os.path.join(cwd, path))
+
+
+def _get_conanfile_path(path, cwd, py):
+    """
+    param py= True: Must be .py, False: Must be .txt, None: Try .py, then .txt
+    """
+    path = _make_abs_path(path, cwd)
+
+    if os.path.isdir(path):  # Can be a folder
+        if py:
+            path = os.path.join(path, "conanfile.py")
+        elif py is False:
+            path = os.path.join(path, "conanfile.txt")
         else:
-            return os.path.abspath(cwd)
-    else:
-        return os.getcwd()
+            path_py = os.path.join(path, "conanfile.py")
+            if os.path.exists(path_py):
+                path = path_py
+            else:
+                path = os.path.join(path, "conanfile.txt")
```

Comment:
Why not` CONANFILE` and `CONANFILE_TXT` here?

Reply:
"conanfile.py" are so widely used in the codebase. ``CONANFILE`` was a very early mechanism when we though we would like to change the recipe name. Now it is so hardcoded in the system, it will never change, so instead of doing an ``import ... CONANFILE`` I prefer to see the file name explicitely. One level less of indirection=easier to read.